### ISSUE TITLE
Post Summary Panel: Change the "Publish" label to a more accurate label

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#63342]
## What problem does this address?

As I understand it, the word "Publish" is only used as a verb. Therefore, I think it is appropriate to leave the blue button in the upper right as "Publish".

![image](https://github.com/WordPress/gutenberg/assets/54422211/5d387e6a-aca0-4d52-9a04-4fbbdb892674)

On the other hand, the same word "Publish" is used in the summary panel. It feels strange that this label is the only one that is a verb, while all the other labels are nouns.

![image](https://github.com/WordPress/gutenberg/assets/54422211/edb0c58e-cc61-4f9d-8310-9a3e192d87d1)

Perhaps the word "Publish" can cover both the meaning of a verb and a noun in English, but this translatable string does not have any additional context.

Therefore, it is not possible to apply a context-sensitive translation in locales other than English.

## What is your proposed solution?

I would like to suggest using a more accurate label other than "Publish" in the summary panel.

If we want to follow the label in the classic editor, "**Published on**" might be good.

![image](https://github.com/WordPress/gutenberg/assets/54422211/198a8dcb-7621-4ce6-ad57-f51dc6b8c707)

Alternatively, I would like to suggest the text "**Published date**".

cc @WordPress/gutenberg-design @afercia

**Comment by afercia** [comment]
@t-hamano good point. I'd agree the verb sounds a bit out of place there. In a way this was also noted in an user session long time ago, see https://github.com/WordPress/gutenberg/issues/470#issuecomment-386804617

> "Publish" is weird to me, it seems like an action but it's just text ...

Couple more considerations:

The default date/time value when creating a new post is `Immediately`, which is likely the reason why this visual label was changed to `Publish`. The new wording should complement well with `Immediately`.

The verb `Publish` is used also as the title of the calendar popover. I'd think it could be improved there as well. Scrsenshot:

![Screenshot 2024-07-10 at 10 23 23](https://github.com/WordPress/gutenberg/assets/1682452/9f791cea-969b-4fa3-b59a-b2597e46eacf)

**Comment by jameskoster** [comment]
Date information is a bit messy. In data views there is a "Date" field which conditionally displays one of the following:

* Last _modified_ date for unpublished posts
* Exact _publish date_ for published posts

Then there's this "Publish" control in the Inspector, which communicates just the publish date for both published and unpublished posts.

Would it make sense to tidy this up a bit? For instance we might store separately the publish date and the modified date. This would be handy in data views because it would allow you to sort with more granularity, e.g. view published posts sorted by last modified. This could also inform the labelling in the Inspector; with two such fields you might expect them to be named "Modified date" and "Publish date". So in the Inspector we'd simply update "Publish" to "Publish date".

Slightly tangential, but I would also pose that the "Publish date" control is not shown at all for draft posts, which would circumvent the awkward "Publish date: Immediately" language. "Publish date" only really makes sense when the post is scheduled, or already published. Otherwise it seems reasonable to assume that a publish action would result in an immediate publish date.

**Comment by afercia** [comment]
Worth noting the Publish date popover [does have](https://github.com/WordPress/gutenberg/blob/6b53293fee26e7ff1c31f037400d80c6b9255c20/packages/editor/src/components/post-schedule/panel.js#L48) an `aria-label="Change publish date"`.

The popover doesn't have an ARIA role though so this is just exposed as a labeled `generic` element. Some screen readers will announce it as `group`, which is arguably correct. Other popovers are labeled as well.

**Comment by afercia** [comment]
A couple more considerations: 

**The adverb 'immediately'**
I don't like that it is not used consistently. In the Panel it's 'immediately' but in the calendar popover is 'Now'. This inconsistency doesn't help users to _immediately_ understand what 'Now' does. I'd argue that whether it's 'immediately' or 'Now', the UI should use the same term everywhere. Screenshot:

![Screenshot 2024-07-25 at 14 26 45](https://github.com/user-attachments/assets/74b70759-f04d-4c05-a7d6-a2cdbe62a0ab)

**The publish date/time label**
I'm not saying that it is necessarily better but I'd like to remind everyone that in Classic editor the visual label for the publish date/time is dynamic and it changes depending whether the selected date / time is in the past, in the future or 'now':

![Screenshot 2024-07-25 at 14 27 36](https://github.com/user-attachments/assets/f0d68927-d927-4ce1-8a12-e834b589d9d3)

**Comment by t-hamano** [comment]
> I'm not saying that it is necessarily better but I'd like to remind everyone that in Classic editor the visual label for the publish date/time is dynamic and it changes depending whether the selected date / time is in the past, in the future or 'now':

This is interesting. Would it make sense to apply this dynamic labeling to the block editor as well?

> but in the calendar popover is 'Now'. This inconsistency doesn't help users to immediately understand what 'Now' does.

Wouldn't it be easier to understand what "Now" means by changing the title of the popover to "Publish date"?

![image](https://github.com/user-attachments/assets/6a59fcfe-5251-44a2-9b9a-9088b2bf7ad4)

Either way, when using the term "Publish" as a label, I believe some context is needed, for example:

```javascript
_x( 'Publish', 'as label' );
```

**Comment by afercia** [comment]
> Wouldn't it be easier to understand what "Now" means by changing the title of the popover to "Publish date"?

That would help.
To me, one of the most weird things is when starting from a publish date / time:

![Screenshot 2024-07-25 at 15 59 37](https://github.com/user-attachments/assets/d4fa9b99-eacb-400a-bbab-47362a7cf2ca)

and then I click 'Now' in the Popover, the date / time changes to 'Immediately':

![Screenshot 2024-07-25 at 15 59 57](https://github.com/user-attachments/assets/3eb30c53-06d8-4df1-8ef1-951dbc39ac67)

as a user, I would expect it to change to 'Now' because, well, I just clicked a button that says 'Now'.
Or, the other way around: they should both be labeled 'Immediately'.

> Either way, when using the term "Publish" as a label, I believe some context is needed,

Good point, the `_x()` translation function would help.

**Comment by jameskoster** [comment]
Now that Status is a dedicated, editable field I'm curious whether we need to display the publish date for unpublished pages. 

Setting a publish date for a draft feels a bit strange because it means the page is no longer a draft; it's scheduled. It's very easy for these two fields to end up in an incompatible state, and there's a lot of code behind the scenes to avoid that happening.

I tend to lean toward intrinsically linking the status & publish date control, something roughly like this:

<img width="1186" alt="Mockup showing the 'Status & visibility' popover, with inline date field" src="https://github.com/user-attachments/assets/0649b59e-1a78-4036-99bd-5ffe222c861a">

This feels a bit more natural to use, and avoids the aforementioned incompatibilities. In the short term we might try conditionally showing/hiding the "Publish" row in the Summary panel.

**Comment by t-hamano** [comment]
When I write a personal blog, I sometimes decide the publish date I want in advance and set the status to Draft until the article is completed. I think there are many other cases where the publish date is decided in advance, such as WordPress release announcement articles.

For this reason, I prefer to have the publication date always displayed.

**Comment by jameskoster** [comment]
Thanks for sharing that context, I can see it's a workflow that makes sense in certain situations.

Changing the "Publish" to "Publish date" and aligning the "Now" and "Immediately" labels seems like a good change to me.

It's probably separate but I don't love the UI/UX for the "Now" button. The interaction with the calendar isn't immediately obvious... I wonder if it should be more connected, and use a more conventional control:

<img width="750" alt="date" src="https://github.com/user-attachments/assets/5e798167-4d50-4d91-ba50-200081a27979">

**Comment by t-hamano** [comment]
> I wonder if it should be more connected, and use a more conventional control:

This is a great idea and I think it could be pursued in parallel with this issue 👍

**Comment by afercia** [comment]
> I don't love the UI/UX for the "Now" button. The interaction with the calendar isn't immediately obvious... I wonder if it should be more connected, and use a more conventional control:

Yes I'd totally agree. The only concern is that if the default is 'Now', to switch to the calendar and set a date I'd have to click one more time. We're also assuming that 'Now' is the most used choice, while it isn't necessarily true. How about let users decide whether to 'stick' to the 'Now' option or to the 'Choose a date' one?

**Comment by jameskoster** [comment]
@afercia you mean make the default selection an editor preference, or a site setting? That seems reasonable, but perhaps a separate endeavour?

@t-hamano do you think this one is ready for dev?

**Comment by afercia** [comment]
> @afercia you mean make the default selection an editor preference, or a site setting? That seems reasonable, but perhaps a separate endeavour?

@jameskoster definitely can be split in a separate issue for broader discussion. I'd tend to think each user may have a personal preference or editorial flow where one of the 'Now' or 'Choose a date' options would be the most used one. As such, I'd tend to think it should be a user preference for the editor. Also, I'd tend to think the control to set the preference should be 'in place' e.g. something like 'Make this the default choice'.
